### PR TITLE
youtubeuploader 21.05 (new formula)

### PR DESCRIPTION
**Pull request by sidneys**
This pull request adds a formula for `youtubeuploader`, a cross-platform command line tool for [YouTube](https://youtube.com) uploads.

The project is hosted on GitHub:
https://github.com/porjo/youtubeuploader

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/youtubeuploader.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Youtubeuploader < Formula
+  desc "Scripted uploads to Youtube"
+  homepage "https://github.com/porjo/youtubeuploader"
+  url "https://github.com/porjo/youtubeuploader/archive/21.05.tar.gz"
+  sha256 "cc087e05e9a31408a6941030cdb933e6e52d619960e765dbb3d91e3661d3dc98"
+  license "Apache-2.0"
+  head "https://github.com/porjo/youtubeuploader.git", branch: "master"
+
+  depends_on "dep" => :build
```
Comment (carlocab) [inline Formula/youtubeuploader.rb:9]: Is there a way to build without this? `dep` is deprecated, so we don't want to use that.

File: Formula/youtubeuploader.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    path = buildpath/"src/github.com/porjo/youtubeuploader"
+    path.install Dir["*"]
+    cd path do
+      system "go", "build", "-a", "-ldflags", "-s -X main.appVersion=#{version}", "-o", "#{bin}/youtubeuploader"
+      prefix.install_metafiles
+    end
```
Comment (carlocab) [inline Formula/youtubeuploader.rb:19]:
```suggestion
    system "go", "build", "-a", *std_go_args(ldflags: "-s -X main.appVersion=#{version}")
```
Does this not work? We don't want to have to set `GOPATH`, as this is no longer needed for building up-to-date Go formulae. What does the `-a` flag do?

File: Formula/youtubeuploader.rb (hunk 1 above)
Comment (sidneys) [reply Formula/youtubeuploader.rb:9]: Yes, the build seems to complete just fine without `dep`.
I'll remove the statement in the next commit.

File: Formula/youtubeuploader.rb (hunk 2 above)
Comment (sidneys) [reply Formula/youtubeuploader.rb]: Yes, this works perfectly too, also without the `-a` flag.

The suggested changes have just been pushed.

**Review by carlocab — APPROVED** [review]
LGTM. Thanks @sidneys!

**Comment by sidneys** [comment]
Great ☺️ Cheers.

**Pull request closed without merging**